lib/sinon/util/fake_xml_http_request.js

```diff
@@ -545,7 +545,7 @@ extend(FakeXMLHttpRequest.prototype, sinonEvent.EventTarget, {
         this.requestHeaders = {};
         this.responseHeaders = {};
 
-        if (this.readyState > FakeXMLHttpRequest.UNSENT && this.sendFlag) {
+        if (this.readyState > FakeXMLHttpRequest.UNSENT && this.sendFlag && this.readyState < FakeXMLHttpRequest.DONE) {
```

Comment:
This could be `this.readyState !== FakeXMLHttpRequest.DONE`. This seems a bit more meaninful to me. This is inline with the readystateCHANGE event since it can't change from 4 to 4. That's not a change and thus it shouldn't fire.